pallets/rmrk-core/src/lib.rs

```diff
@@ -354,18 +356,45 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin.clone())?;
 
+			// Get current owner for child removal later
+			let parent =
+				pallet_uniques::Pallet::<T>::owner(collection_id, nft_id);
+			// Check if parent returns None which indicates the NFT is not available
+			ensure!(parent.is_some(),Error::<T>::NoAvailableNftId);
+			let current_owner = parent.as_ref().unwrap();
+
 			let max_recursions = T::MaxRecursions::get();
-			Self::nft_send(
+			let new_owner_account = Self::nft_send(
 				sender.clone(),
 				collection_id,
 				nft_id,
 				new_owner.clone(),
 				max_recursions,
 			)?;
 
+			pallet_uniques::Pallet::<T>::do_transfer(
+				collection_id,
+				nft_id,
+				new_owner_account.clone(),
+				|class_details, details|
+					Ok(())
+			)?;
+
+			// Handle Children StorageMap for NFTs
+			let current_cid_nid =  Pallet::<T>::decode_nft_account_id::<T::AccountId>(current_owner.clone());
+			if current_cid_nid.is_some() {
+				// remove child from parent
+				Pallet::<T>::remove_child((current_cid_nid.unwrap().0, current_cid_nid.unwrap().1), (collection_id, nft_id));
+			}
+			// add child to new parent if NFT virtual address
+			let new_cid_nid = Pallet::<T>::decode_nft_account_id::<T::AccountId>(new_owner_account.clone());
+			if new_cid_nid.is_some() {
```

Comment:
if let Some(new_cid_nft) = new_cid_nft { ... }

Then you can use the tuple as the arg directly without unwrap